packages/fxa-auth-server/scripts/transfer-users/apple.js

```diff
@@ -17,24 +17,39 @@ const GRANT_TYPE = 'client_credentials';
 const SCOPE = 'user.migration';
 const USER_MIGRATION_ENDPOINT = 'https://appleid.apple.com/auth/usermigrationinfo';
 
-const APPLE_PROVIDER = 2;
+const APPLE_PROVIDER = 'apple';
 
 export class AppleUser {
-  constructor(email, transferSub, uid, alternateEmails, db, writeStream, config) {
+  constructor(email, transferSub, uid, alternateEmails, db, writeStream, config, mock) {
     this.email = email;
     this.transferSub = transferSub;
     this.uid = uid;
     this.alternateEmails = alternateEmails || [];
     this.db = db;
     this.writeStream = writeStream;
     this.config = config;
+    this.mock = mock;
   }
 
   // Exchanges the Apple `transfer_sub` for the user's profile information and
   // moves the user to the new team.
   // Ref: https://developer.apple.com/documentation/sign_in_with_apple/bringing_new_apps_and_users_into_your_team#3559300
   async exchangeIdentifiers(accessToken) {
     try {
+      
+      if (this.mock) {
```

Comment:
Instead of using a flag, another way of doing this is just overriding the functionality as needed. This can be down by creating an AppUserMock class, or directly mocking class methods with a testing library.  Doing it that way avoids adding 'mocking' logic inside the actual method implementations. If this approach is difficult it might be a sign that the methods in the class could be further broken down into smaller protected methods.

Reply:
Fair point, I probably won't change this for now since hoping to do a test run today. We wouldn't do a real prod run until July, so should be able to tweak it if needed.